reana_client/utils.py

```diff
@@ -191,15 +191,27 @@ def _validate_environment(reana_yaml):
         workflow_steps = reana_yaml["workflow"]["specification"]["steps"]
         _validate_serial_workflow_environment(workflow_steps)
     elif workflow_type == "yadage":
-        logging.warning(
-            "The environment validation is not implemented for Yadage workflows yet."
-        )
+        with open(reana_yaml["workflow"].get("file")) as f:
+            yadage_steps = yaml.load(f.read(), Loader=yaml.FullLoader)
+        _validate_yadage_workflow_environment(yadage_steps["stages"])
     elif workflow_type == "cwl":
         logging.warning(
             "The environment validation is not implemented for CWL workflows yet."
         )
 
 
+def _validate_yadage_workflow_environment(workflow_steps):
+    """Validate environments in REANA yadage workflow.
+
+    :param workflow_steps: List of dictionaries which represents different steps involved in workflow.
+    :raises Warning: Warns user if the workflow environment is invalid in yadage workflow steps.
+    """
+    for stage in workflow_steps:
+        environment = stage["scheduler"]["step"]["environment"]
+        image = "{}:{}".format(environment["image"], environment["imagetag"])
```

Comment:
@mvidalgarcia I combined the image name and image tag here to send it to `_validate_image_tag`.

Should we go ahead with creating a separate function for that?

Reply:
Something like what you have should already be fine. We can look later to avoid some repetition if possible.